### PR TITLE
Atomic design refactoring

### PR DESCRIPTION
**Pull request by fragsalat**
*NOTE*: DON'T MERGE YET

This PR is the first step to migrate to an atomic design approach for our scss repo. I removed all things I could not find any usage for and which are not styles yet. There are some todo's left like:
- Clarifying and resolving open @todo's (Please comment on all todo's with your opinoin even if it just says that it can stay as it is)
- Style checkbox and radio box again because the icon svg's are finally removed
- Check migration of all projects
- Check if there can be a better structuring (Like grid to atoms or icons to utils or something like this)

@wholesale-design-system/engineers

### CODE REVIEW

**Comment by faboweb** [comment]
Just a remark: This will probably break the components repo.

**Comment by fragsalat** [comment]
This will break everything. Therefore I said pls don't merge it. ^^ If everyone is fine with it I'm taking care to migrate the components and demo page to this new structure.

**Comment by faboweb** [comment]
👍

**Comment by schluchter** [comment]
You're a hero.

File: utils/variables/_layers.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @todo where comes this values from, do and why does they make sense?
```
Comment (tonysaad) [inline utils/variables/_layers.scss]: Values come from here: https://www.sitepoint.com/better-solution-managing-z-index-sass/
Some cases you'd need -ve values especially with opacity (However we don't have such a case yet)

File: utils/_helpers.scss [hunk 2]
```diff
@@ -68,39 +68,10 @@ $debug: false !default;
...
-      }
-
-      $i: $i + 2;
-    }
-    @media #{$media-query} {
-      @content;
-      $grid-columns: $default-grid-columns !global;
-    }
+// @todo
+@mixin media($query) {
```
Comment (tonysaad) [inline utils/_helpers.scss]: Would be great to have a temporarily solution that makes components runs well
screen shot 2017-03-31 at 14 09 48

ex.
```
$breakpoints: (
  small: 767px,
  medium: 992px,
  large:  1200px
) !default;
```

```
@mixin media($breakpoint) {
  // If the key exists in the map
  @if map-has-key($breakpoints, $breakpoint) {
    // Prints a media query based on the value
    @media (min-width: map-get($breakpoints, $breakpoint)) {
      @content;
    }
  }
 
  // If the key doesn't exist in the map
  @else {
    @warn "Unfortunately, no value could be retrieved from `#{$breakpoint}`. "
        + "Available breakpoints are: #{map-keys($breakpoints)}.";
  }
}
```

File: utils/_variables.scss [hunk 3]
```diff
@@ -1,188 +1,6 @@
...
-//Input Variables
-$input-border-color: $color-border-input-default;
-$input-shadow-color: lighten($input-border-color, 15%);
-$input-shadow-inset: inset 0px 1px 3px 0px $input-shadow-color;
-
-$default-feature: min-width;
-$column: 90px;
-$gutter: 30px;
-$grid-columns: 12;
-$max-width: 1200px;
```
Comment (fragsalat) [inline utils/_variables.scss]: Do you use outer-container mixin? 
I though we should use the grid for this

File: utils/variables/_layers.scss (hunk 1 above)
Comment (fragsalat) [inline utils/variables/_layers.scss]: Thx I'll check it and remove the todo

File: utils/_helpers.scss (hunk 2 above)
Comment (fragsalat) [inline utils/_helpers.scss]: Lets sit together with ux to define those clearly to don't produce unused code again :)

File: utils/_variables.scss (hunk 3 above)
Comment (tonysaad) [inline utils/_variables.scss]: Only used in the current ws-header.
Yes right, we should take out in the new header and just use the grid

**Comment by fragsalat** [comment]
@tonysaad added the discussed break points. If you are fine with them pls press the merge button :)

I'll provide a PR to adopt the changes within the components repo

**Comment by tonysaad** [comment]
@fragsalat Somehow my <ul>s and <li>s all have default margins, padding and list-style-type I'm not sure why, maybe it has been removed from the reset file.
I added this in my project, but I think we should add it our _reset too, what do you think ?
`ul, li {
  margin: 0;
  padding: 0;
  list-style-type: none;
}`

**Comment by fragsalat** [comment]
:+1:  Should definitely be part of the reset file

**Pull request merged**